docs/DeveloperGuide.adoc

```diff
@@ -479,3 +479,81 @@ Note: If the format of `visit-now-update` arguments are wrong, the error message
 .. Prerequisite: There is no ongoing visit.
 .. Test case: `visit-end` +
    Expected: No visit is finished. An error message will appear in the status message.
+
+=== Adding an appointment
+
+. Adding an appointment while all appointments are listed
+
+.. Prerequisites:
+... List all appointments using the `appt-list` command. Multiple appointments in the list.
+... Ensure all the suggested date times below don't already exist in the appointment book. If they do, try with start and end date times that don't exist in the appointment list to get the expected results.
+.. Test case: `appt-add p/1 sdt/01-01-2020 1000 edt/01-01-2020 1200` +
+   Expected: Appointment is added to the list (based on its start date and time as the list is sorted by that) with first patient's details from the list of patients and the input start and end date times. Details of the added appointment shown in the status message. Timestamp in the status bar is updated.
+.. Test case: `appt-add p/1 sdt/14-02-2020 2000 edt/14-02-2020 2100 rmon/4` +
+   Expected: Appointment is added to the list with first patient's details from the list of patients, the input start and end date times, and recurring frequency of 4 months. The added appointment will be marked `recurring` as well. Details of the added appointment shown in the status message. Timestamp in the status bar is updated.
+.. Test case (two commands): `appt-add p/1 sdt/12-02-2020 2000 edt/12-02-2020 2100 ryr/1` +
+                             Followed by: +
+                             `appt-add p/1 sdt/12-02-2020 2000 edt/12-02-2020 2100 rmon/1` +
+   Expected: No appointment is added. Error details shown in the status message. Status bar remains the same.
+.. Test case (two commands): `appt-add p/1 sdt/10-02-2020 2000 edt/12-02-2020 2100` +
+                             Followed by: +
+                             `appt-add p/1 sdt/10-02-2020 2000 edt/12-02-2020 2100` +
+   Expected: No appointment is added. Error details shown in the status message. Status bar remains the same.
+.. Other incorrect appointment delete commands to try: `appt-add`, `appt-delete p/x` (where x is larger than the patient list size) +
+   Expected: Similar to previous.
+
+=== Editing an appointment
+
+. Editing an appointment while all appointments are listed
+
+.. Prerequisites:
+... List all appointments using the `appt-list` command. Multiple appointments in the list.
+... Ensure all the suggested date times below don't already exist in the appointment book. If they do, try with start and end date times that don't exist in the appointment list to get the expected results.
```

Comment:
index of appointment must be present is a preq too

Reply:
I think by prerequisites they mean the state application, and in this case that of the appointment list, before the command is entered. 

Index of appointment not being after the command word is a test case that throws an error (have mentioned further down).